app/cdap/components/PipelineDiff/DiffList/index.tsx

```diff
@@ -50,73 +89,76 @@ export const DiffList = () => {
-            getAvailabePluginsMapKeyFromPlugin(fromStage.plugin)
-          );
-          const fromIconName = getPluginIcon(getPluginNameFromStageDiffKey(fromStageKey));
+  const pluginDiffList = useMemo(() => {
+    return Object.keys(diffMap.stages)
+      .map((diffKey) => {
+        // The stage must exist in either the second pipeline(current version)
+        // or the first pipeline (older version).
+        const stage = diffMap.stages[diffKey].stage2 ?? diffMap.stages[diffKey].stage1;
+        return getStageProps(stage, availablePluginsMap, diffMap);
+      })
+      .filter(({ nodeName }) => nodeName.toLowerCase().includes(search.toLowerCase()));
+  }, [search, diffMap, availablePluginsMap]);
 
-          const toNodeName = getStageNameFromStageDiffKey(toStageKey);
-          const toCustomIconSrc = getCustomIconSrc(
-            availablePluginsMap,
-            getAvailabePluginsMapKeyFromPlugin(toStage.plugin)
-          );
-          const toIconName = getPluginIcon(getPluginNameFromStageDiffKey(toStageKey));
-          return (
-            <ConnectionDiffListItem
-              fromNodeName={fromNodeName}
-              fromCustomIconSrc={fromCustomIconSrc}
-              fromIconName={fromIconName}
-              toNodeName={toNodeName}
-              toCustomIconSrc={toCustomIconSrc}
-              toIconName={toIconName}
-              diffKey={connectionKey}
-              onClick={() => dispatch(actions.startNavigateTo(connectionKey))}
-              diffType={diffMap.connections[connectionKey].diffIndicator}
-              key={connectionKey}
-            />
-          );
-        })}
-      </List>
-    </DiffListRoot>
+  const connectionDiffList = useMemo(() => {
+    return Object.keys(diffMap.connections)
+      .map((connectionKey) => {
+        const fromStage = diffMap.connections[connectionKey].from;
+        const fromProps = getStageProps(fromStage, availablePluginsMap, diffMap);
+        const toStage = diffMap.connections[connectionKey].to;
+        const toProps = getStageProps(toStage, availablePluginsMap, diffMap);
+        return {
+          fromNodeName: fromProps.nodeName,
+          fromCustomIconSrc: fromProps.customIconSrc,
+          fromIconName: fromProps.iconName,
+          toNodeName: toProps.nodeName,
+          toCustomIconSrc: toProps.customIconSrc,
+          toIconName: toProps.iconName,
+          diffKey: connectionKey,
+          diffIndicator: diffMap.connections[connectionKey].diffIndicator,
+        };
+      })
+      .filter(
+        ({ fromNodeName, toNodeName }) =>
+          fromNodeName.toLowerCase().includes(search.toLowerCase()) ||
+          toNodeName.toLowerCase().includes(search.toLowerCase())
+      );
+  }, [search, diffMap, availablePluginsMap]);
+
+  return (
+    <DiffSidebar elevation={3}>
+      <DiffSearch search={search} setSearch={setSearch} />
+      <DiffListContainer>
+        <DiffListRoot dense={true} style={{ backgroundColor: 'inherit' }}>
```

Comment:
If you are not using DiffListRoot elsewhere, can you put the style into its styled component

Reply:
It was already in the styled component, deleted the extra style prop.